src/anamoly_detection/equation_parser.py

```diff
@@ -0,0 +1,38 @@
+def parse_equation(coefficients: str):
+    """
+    Create a callable function representing an equation
+    Function in form of:
+
+        "a, b, ... , z" -> f(x) = (a * x) ... (y * x ^ ?) + z
+                                 ...
+        "a, b"          -> f(x) = (a * x) + b
+        "a"             -> f(x) = a
+
+    :param coefficients: CSV in string form of the coefficients
+    :return: A callable function that calculates a polynomial of variable degree, based on given coefficients
+    """
+    # Equations exists as coefficients to a (len(coefficients) - 1) degree polynomial
+    list_of_coefficients = coefficients.replace(" ", "").split(",")
+    list_of_expressions = []
+    degree = len(list_of_coefficients)
+    current_degree = 1
+
+    # Escape on no coefficients
+    if degree == 0 or list_of_coefficients[0] == '':
+        return lambda x: 0
+
+    # Loop through coefficients, assigning them to their respective expressions based on degree
+    for coefficient in list_of_coefficients:
+        if coefficient == "0":
+            # No need to waste time with 0 value coefficient
+            continue
+        if current_degree == degree:
+            # Final value is always a constant (C)
+            list_of_expressions.append(lambda x, cof=coefficient: float(cof))
+        else:
+            # [0:-2] coefficients are normal polynomials to the power of the current_degree
+            list_of_expressions.append(lambda x, cof=coefficient, deg=current_degree: float(cof) * pow(x, deg))
+        current_degree += 1
+
+    # Return a function that sums up the value of each expression
+    return lambda x: sum(list(map(lambda y: y(x), list_of_expressions)))
```

Comment:
Why is this function is in a standalone file without a class definition or other functions that utilize it? It can probably be moved to an existing file instead.

Reply:
I personally dislike large files that contain too much code, but I'll move it if that's what you want